### PR TITLE
resources: add DT job enums and types

### PR DESCRIPTION
**Pull request by wexxlee**

### CODE REVIEW

**Comment by valarnin** [comment]
Were the PCT/VIP job IDs (41, 42) verified, or just assumed?

**Comment by wexxlee** [comment]
> Were the PCT/VIP job IDs (41, 42) verified, or just assumed?

Verified in `ClassJob`.

File: resources/party.ts [hunk 1]
```diff
@@ -396,6 +396,12 @@ const jobLocalizedAbbr: Record<Job, LocaleText> = {
     cn: '贤者',
     ko: '현자',
   },
+  VPR: {
+    en: 'VPR',
```
Comment (MaikoTan) [inline resources/party.ts:400]:
```suggestion
    en: 'VPR',
    de: 'VPR',
    fr: 'VPR',
    ja: 'ヴァイパー',
```

File: resources/party.ts [hunk 2]
```diff
@@ -396,6 +396,12 @@ const jobLocalizedAbbr: Record<Job, LocaleText> = {
...
+  },
+  PCT: {
+    en: 'PCT',
```
Comment (MaikoTan) [inline resources/party.ts:403]:
```suggestion
    en: 'PCT',
    de: 'PKT',
    fr: 'PIC',
    ja: 'ピクトマンサー',
```

File: resources/party.ts [hunk 3]
```diff
@@ -727,6 +733,12 @@ const jobLocalizedFull: Record<Job, LocaleText> = {
     cn: '贤者',
     ko: '현자',
   },
+  VPR: {
+    en: 'Viper',
```
Comment (MaikoTan) [inline resources/party.ts:737]:
```suggestion
    en: 'Viper',
    de: 'Viper',
    fr: 'Rôdeur vipère',
    ja: 'ヴァイパー',
```

File: resources/party.ts [hunk 4]
```diff
@@ -727,6 +733,12 @@ const jobLocalizedFull: Record<Job, LocaleText> = {
...
+  },
+  PCT: {
+    en: 'Pictomancer',
```
Comment (MaikoTan) [inline resources/party.ts:740]:
```suggestion
    en: 'Pictomancer',
    de: 'Piktomant',
    fr: 'Pictomancien',
    ja: 'ピクトマンサー',
```

**Review by MaikoTan — APPROVED** [review]
Good, I add translations from the game files extracted

**Comment by MaikoTan** [comment]
Oops, didn't notice that there is a merge queue.

**Pull request merged**